core/src/main/scala/org/apache/spark/deploy/ExternalShuffleService.scala

```diff
@@ -37,15 +39,17 @@ import org.apache.spark.util.Utils
  * Optionally requires SASL authentication in order to read. See [[SecurityManager]].
  */
 private[deploy]
-class ExternalShuffleService(sparkConf: SparkConf, securityManager: SecurityManager)
+class ExternalShuffleService(
+    sparkConf: SparkConf,
+    securityManager: SecurityManager,
+    transportConf: TransportConf,
+    blockHandler: ExternalShuffleBlockHandler)
```

Comment:
I would do this a different way, just define a
```
protected def newBlockHandler(): ExternalShuffleBlockHandler = {
  new ExternalShuffleBlockHandler(transportConf)
}
```
and then have the subclass override it